resources/sdk/purecloudjavascript/templates/configuration.mustache

```diff
@@ -0,0 +1,167 @@
+const ConfigParser = require('configparser');
+const os = require('os');
+const path = require('path');
+const fs = require('fs');
+
+import Logger from './logger.js';
+
+class Configuration {
+	/**
+	 * Singleton getter
+	 */
+	get instance() {
+		return Configuration.instance;
+	}
+
+	/**
+	 * Singleton setter
+	 */
+	set instance(value) {
+		Configuration.instance = value;
+	}
+
+	constructor() {
+		if (!Configuration.instance) {
+			Configuration.instance = this;
+		}
+
+		this.configPath = typeof window !== 'undefined' ? '' : path.join(os.homedir(), '.genesyscloudjavascript', 'config');
+		this.refresh_access_token = true;
+		this.refresh_token_wait_max = 10;
+		this.live_reload_config = true;
+		this.host;
+		this.environment;
+		this.basePath;
+		this.authUrl;
+		this.config;
+		this.logger = new Logger();
+		this.setEnvironment();
+		this.liveLoadConfig();
+	}
+
+	liveLoadConfig() {
+		// If in browser, don't read config file, use default values
+		if (typeof window !== 'undefined') {
+			this.configPath = '';
+			return;
+		}
+
+		this.updateConfigFromFile();
+
+		if (this.live_reload_config && this.live_reload_config === true) {
+			try {
+				const watcher = fs.watchFile(this.configPath, { persistent: false, interval: 30000 }, (eventType, filename) => {
+					this.updateConfigFromFile();
+					if (!this.live_reload_config) {
+						watcher.close();
```

Comment:
I got the following error after setting live_reload_config to false in the config file:
TypeError: watcher.close is not a function

Reply:
I couldn't reproduce the error on my end. Tried several combinations and all worked. Let me know if it's still not working on your side, then we can chat and use some specific examples to flush out the bug.

Reply:
From looking at the docs, fs.watch returned an FSWatcher which has the close() method to stop watching.
fs.watchFile returns nothing and the following function is used to stop watching:
`unwatchFile(filename: PathLike, listener?: (curr: Stats, prev: Stats) => void):`

Reply:
yep changed it to use `unwatchFile()` to stop live loading